lms/djangoapps/grades/models.py

```diff
@@ -47,12 +47,17 @@ def to_json(self):
         if self._json is None:
             sorted_blocks = sorted(self, key=attrgetter('locator'))
             list_of_block_dicts = [block._asdict() for block in sorted_blocks]
+            course_key = next(iter(self)).locator.course_key  # all blocks are from the same course
```

Comment:
We should add a test case for subsections without any problem blocks.  Not sure what we are currently doing in that case.  Do we still persist a grade of 0/0 for those subsections?